ironic_prometheus_exporter/messaging.py

```diff
@@ -18,13 +19,32 @@ class PrometheusFileDriver(notifier.Driver):
     """Publish notifications into a File to be used by Prometheus"""
 
     def __init__(self, conf, topics, transport):
-        self.file_path = conf.oslo_messaging_notifications.file_path
-        if not self.file_path.endswith('.json'):
-            raise Exception('The file should end with .json')
-        if not os.path.exists(os.path.dirname(self.file_path)):
-            os.makedirs(os.path.dirname(self.file_path))
+        self.files_dir = conf.oslo_messaging_notifications.files_dir
+        if not os.path.exists(self.files_dir):
+            os.makedirs(self.files_dir)
         super(PrometheusFileDriver, self).__init__(conf, topics, transport)
 
     def notify(self, ctxt, message, priority, retry):
-        with open(self.file_path, 'w') as prometheus_file:
-            json.dump(message, prometheus_file)
+        try:
+            node_parser_manager = manager.ParserManager(message)
+            node_metrics = node_parser_manager.merge_information()
+            node_name = message['payload']['node_name']
+            node_file = open(os.path.join(self.files_dir, node_name), 'w')
+            node_file.write(node_metrics)
+            node_file.close()
+        except Exception as e:
+            print(e)
+
+
+class SimpleFileDriver(notifier.Driver):
```

Comment:
Why do we need this notifier?

Reply:
I added it so we can collect metrics samples from any source (Redfish or others) since we don't know how they will look like

Reply:
Can we have a notifier per node? I doubt it. Then you need one notifier to be able to handle all possible drivers more or less gracefully.

Reply:
I know, the Simple driver is just used to collect the data, after have the data we will update the Prometheus Driver it will call the manager that would handle the correct parser for the metrics and so on.